manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -160,19 +174,24 @@ public Long getProcessorsCount(String bridgeId, String customerId) {
         return processorDAO.findByBridgeIdAndCustomerId(bridge.getId(), bridge.getCustomerId(), queryInfo);
     }
 
-    @Transactional
     @Override
     public void deleteProcessor(String bridgeId, String processorId, String customerId) {
-        Processor processor = processorDAO.findByIdBridgeIdAndCustomerId(processorId, bridgeId, customerId);
-
-        ProcessorDefinition processorDefinition = jsonNodeToDefinition(processor.getDefinition());
-        ActionProvider actionProvider = actionProviderFactory.getActionProvider(processorDefinition.getRequestedAction().getType());
-
-        connectorService.deleteConnectorIfNeeded(processorDefinition.getResolvedAction(), processor, actionProvider);
+        List<ConnectorEntity> connectorEntitiesToBeDeleted = updateProcessorConnectorForDeletionOnDB(bridgeId, processorId, customerId);
+        connectorEntitiesToBeDeleted.forEach(c -> {
+            LOGGER.info("Firing deletion event for entity: " + c);
+            eventBus.requestAndForget(Events.CONNECTOR_DELETED_EVENT, c);
+        });
+    }
 
+    @Transactional
+    public List<ConnectorEntity> updateProcessorConnectorForDeletionOnDB(String bridgeId, String processorId, String customerId) {
```

Comment:
Is there any reason why this needs to be a separate method to `deleteProcessor`?

Reply:
We don't want to fire the event on the service bus while being in a transaction, so the method that writes on the DB is separated with its own `@Transactional` annotation, when it returns the event is fired